lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,326 @@
+})
+
+SWEP.Primary.Ammo = "Battery"
+SWEP.Primary.ClipSize = 100
+SWEP.Primary.DefualtClip = 100
+SWEP.Primary.Automatic = true
+
+SWEP.Secondary.Ammo = "none"
+SWEP.Secondary.ClipSize = -1
+SWEP.Secondary.DefaultClip = -1
+SWEP.Secondary.Automatic = false
+
+SWEP.ViewModel = "models/brian/laserpointer.mdl"
+SWEP.WorldModel = "models/brian/laserpointer.mdl"
+
+SWEP.laserorgpos = Vector(0,0,0)
+SWEP.laserorgang = Angle(0,0,0)
+SWEP.LaserBeamOn = false
+SWEP.LaserPower = 0
+SWEP.LaserColors = {
+	Color(255,0,0,255),
+	Color(0,255,0,255),
+	Color(255,0,255,255)
+}
+
+if CLIENT then
+	hook.Add("HUDPaint", "BatteryBar", function()
+		local wep = LocalPlayer():GetActiveWeapon()
+		if IsValid(wep) and wep:GetClass() == "weapon_laserpointer" then
+			draw.RoundedBox(8, (ScrW()/2)-64, ScrH()-48, 128, 32, Color(0, 0, 0, 128))
+			draw.RoundedBox(0, (ScrW()/2)-58, ScrH()-42, 1.16*wep:Clip1(), 20, wep.LaserColors[wep.LaserPower+1])
+			draw.DrawText("Battery", "DermaLarge", (ScrW()/2), ScrH()-84, Color(255, 255, 255, 255), TEXT_ALIGN_CENTER)
+		end
+	end)
+	hook.Add("HUDPaintBackground", "BlindnessDraw", function()
+		local ply = LocalPlayer()
+		if IsValid(ply) then
+			surface.SetDrawColor(255,255,255,ply:GetNWFloat("Blindness",0))
+			surface.DrawRect(0, 0, ScrW(), ScrH())
+		end
+	end)
+	hook.Add("PostDrawTranslucentRenderables", "DrawAllLasers", function()
+		for i,v in next, player.GetAll() do
+			local wep = v:GetActiveWeapon()
+			if IsValid(wep) and wep:GetClass() == "weapon_laserpointer" and wep:GetNWBool("LaserBeamOn") and wep:GetNWBool("CanDrawLaser") and v ~= LocalPlayer() then
+				local tr = v:GetEyeTrace()
+				local lsrclr = wep.LaserColors[v:GetNWFloat("LaserPower", 0)+1]
+
+				cam.Start3D(EyePos(), EyeAngles())
+					--local maxbounce = maxbounce or 0
+					local size = math.random()*10
+
+					render.SetMaterial(Material("sprites/bluelaser1"))
+					render.DrawBeam(wep:GetNWVector("laserorgpos",Vector(0,0,0,0))+wep:GetNWVector("laserorgang",Angle(0,0,0)):Up()*6, tr.HitPos, 2, 0, 12.5, lsrclr, false)
+					render.SetMaterial(Material("sprites/light_glow02_add"))
+					render.DrawQuadEasy(tr.HitPos, tr.HitNormal, size, size, lsrclr)
+
+					--[[if tr.Entity:GetMaterial() == "func_reflective_glass" and maxbounce <= 1 then--this is for the refelcted laser
+						local vec = origin-tr.HitPos
+						local normal = tr.HitNormal
+						local reflect = -2*vec:Dot(normal)*normal+vec
+						local trc = util.TraceLine({
+							start = tr.HitPos,
+							endpos = tr.HitPos + reflect:Angle():Forward() * 10000,
+							filter = false
+						})
+
+						render.SetMaterial(Material("sprites/bluelaser1"))
+						render.DrawBeam(tr.HitPos, trc.HitPos, 2, 0, 12.5, lsrclr, false)
+						render.SetMaterial(Material("sprites/light_glow02_add_noz"))
+						render.DrawSprite(trc.HitPos, size, size, lsrclr)
+
+						maxbounce = maxbounce+1
+					end]]
+				cam.End3D()
+			end
+		end
+	end)
+	hook.Add("KeyRelease","RemoveBeam", function(ply, key)

```

Comment:
This copy pasted function should just be outside the `if CLIENT then ... else ... end`

Reply:
oh okay